### PR TITLE
Deploy on OpenShift with dedicated SCC

### PR DESCRIPTION
**Pull request by synarete**
Refactor deployment over OpenShift cluster using custom `samba` SCC. User is expected to deploy the `samba` SCC prior to the deployment of the samba-operator (or, via kustomization mechanism). 

Reconcile ServiceAccount, Role and RoleBinding as part of SmbShare reconcile loop, within the namespace in which the smbshare resides.

### CODE REVIEW

File: internal/conf/config.go
Comment (phlogistonjohn) [inline internal/conf/config.go:16]: This is super-duper nitpicky, and I clearly didn't notice it last time, but shouldn't it be spelled `ClusterTypeOpenShift`?

**Comment by phlogistonjohn** [comment]
I made a few small comments on my first skim. I will def. need to re-read this for a proper review. Please ping me in a few work days if you don't see a full review by then.

File: internal/resources/pods.go
Comment (synarete) [reply internal/resources/pods.go:858]: https://sdk.operatorframework.io/docs/best-practices/pod-security-standards/

File: internal/conf/config.go
Comment (synarete) [reply internal/conf/config.go:16]: ACK. Will fix

**Comment by synarete** [comment]
> Either in this PR or in a follow up I think it would be good to have a new document in the docs dir that goes over some of the important differences between plain k8s and openshift, I think the most important would be to explain that: if you are using openshift and want to create smbshares in namespaces other than the default (which we've already updated via kustomize) the creator of the ns also needs to set the openshift specific security properties.
> 
> (I'm assuming I got the summary right, feel free to correct me)

I am working on samba-on-openshift doc. Will come in a separate PR.

**Comment by phlogistonjohn** [comment]
I want to try microshift later this week and if it goes OK then I will test this PR on it. If I can't get microshift running either I will probably just approve, but if this goes unreviewed past next tues (03-15) feel free to ping me and remind me I owe you a proper review.

**Review by phlogistonjohn — APPROVED** [review]
I built and ran this while following the docs at PR #295. As I now have looked the code over and now have run it, I can strongly say LGTM.

**Review by spuiuk — APPROVED** [review]
Ack

**Pull request merged**
